src/wallet/wallet.cpp

```diff
@@ -922,18 +922,18 @@ CWalletTx* CWallet::AddToWallet(CTransactionRef tx, const CWalletTx::Confirmatio
         wtx.nTimeSmart = ComputeTimeSmart(wtx);
         AddToSpends(hash, &batch);
 
-        std::vector<std::pair<const CTransactionRef&, unsigned int>> outputs;
-        for(unsigned int i = 0; i < wtx.tx->vout.size(); ++i) {
+        std::set<COutPoint> candidates;
+        for (unsigned int i = 0; i < wtx.tx->vout.size(); ++i) {
             if (IsMine(wtx.tx->vout[i]) && !IsSpent(hash, i)) {
                 setWalletUTXO.insert(COutPoint(hash, i));
-                outputs.emplace_back(wtx.tx, i);
+                candidates.emplace(hash, i);
             }
         }
         // TODO: refactor duplicated code between CWallet::AddToWallet and CWallet::AutoLockMasternodeCollaterals
-        if (m_chain) {
-            for (const auto& outPoint : m_chain->listMNCollaterials(outputs)) {
-                LockCoin(outPoint, &batch);
-            }
+        std::vector<COutPoint> protx_utxos;
+        ListProTxCoins(candidates, /*output=*/protx_utxos);
+        for (const auto& utxo : protx_utxos) {
```

Comment:
can be re-written much shorter (see other comments also):

```diff
-        std::vector<COutPoint> protx_utxos;
-        ListProTxCoins(candidates, /*output=*/protx_utxos);
-        for (const auto& utxo : protx_utxos) {
+        for (const auto& utxo : ListProTxCoins(candidates)) {
```

Reply:
Migrated to result-by-return from result-by-ref, resolved in latest push!